ml-agents/mlagents/trainers/torch/utils.py

```diff
@@ -130,24 +131,39 @@ def get_encoder_for_obs(
         :param h_size: Number of hidden units per layer.
         :param vis_encode_type: Type of visual encoder to use.
         """
-        if len(shape) == 1:
-            # Case rank 1 tensor
-            return (VectorInput(shape[0], normalize), shape[0])
-        if len(shape) == 3:
-            ModelUtils._check_resolution_for_encoder(
-                shape[0], shape[1], vis_encode_type
-            )
+        shape = obs_spec.shape
+        dim_prop = obs_spec.dimension_property
+
+        # VISUAL
+        valid_visual = (
```

Comment:
Maybe make these constants somewhere else, or static properties of the DimensionProperty enum? You could also use a `frozenset` so that the checks would be `if dim_prop in ...` instead.